src/faction_camp.cpp

```diff
@@ -514,6 +514,102 @@ static bool update_time_fixed( std::string &entry, const comp_list &npc_list,
     return avail;
 }
 
+bool extract_and_check_orientation_flags( const recipe_id recipe,
+        const point &dir,
+        bool &mirror_horizontal,
+        bool &mirror_vertical,
+        int &rotation,
+        const std::string base_error_message,
+        const std::string actor )
+{
+    mirror_horizontal = recipe->has_flag( "MAP_MIRROR_HORIZONTAL" );
+    mirror_vertical = recipe->has_flag( "MAP_MIRROR_VERTICAL" );
+    rotation = 0;
+    std::string dir_string;
+
+    const auto check_rotation = [&]( const std::string & flag, int rotation_value ) {
+        if( recipe->has_flag( flag ) ) {
+            if( rotation != 0 ) {
+                popup( _( ( base_error_message +
+                            ", the blueprint specifies multiple concurrent rotations, which is not supported" ).c_str() ),
+                       actor,
+                       recipe->get_blueprint() );
```

Comment:
```suggestion
                debugmsg( "%s, the blueprint specifies multiple concurrent rotations, which is not supported",
                          string_format( base_error_message, actor, recipe->get_blueprint() ) );
```
Error message should be displayed with `debugmsg` and does not need translation. (By the way, `_()` has no effect on dynamically generated text. It only works when the enclosed text is a string literal.)